x-pack/legacy/plugins/apm/public/components/shared/MetadataTable/__test__/MetadataTable.test.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import 'jest-dom/extend-expect';
+import { render, cleanup } from 'react-testing-library';
+import { MetadataTable } from '..';
+import {
+  expectTextsInDocument,
+  expectTextsNotInDocument
+} from '../../../../utils/testHelpers';
+import { Transaction } from '../../../../../typings/es_schemas/ui/Transaction';
+
+describe('MetadataTable', () => {
+  afterEach(cleanup);
+  describe('required sections', () => {
+    it('shows message if no data is available', () => {
+      const sections = [
+        {
+          key: 'foo',
+          label: 'Foo',
+          required: true
+        }
+      ];
+      const output = render(
+        <MetadataTable item={{} as Transaction} sections={sections} />
+      );
+      expectTextsInDocument(output, ['Foo', 'No data available']);
+    });
+    it('shows message if property is not available', () => {
+      const sections = [
+        {
+          key: 'foo',
+          label: 'Foo',
+          required: true,
+          properties: ['bar']
+        }
+      ];
+      const item = ({
+        foo: {
+          foobar: 'bar'
+        }
+      } as unknown) as Transaction;
+
+      const output = render(<MetadataTable item={item} sections={sections} />);
+      expectTextsInDocument(output, ['Foo', 'No data available']);
+    });
+  });
+  describe('not required sections', () => {
+    it('does not show section when any item is provided', () => {
```

Comment:
Perhaps:
```suggestion
    it('does not show section when no items are provided', () => {
```

Reply:
A rule of thumb I use is to break things down into another `describe` if I'm using `if` or `when` in an `it` block, so `describe('when no items are provided'...it('does not show the section'...`